.github/workflows/shellcheck.yaml

```diff
@@ -0,0 +1,18 @@
+name: ShellCheck
+
+# See https://github.com/koalaman/shellcheck/wiki
+# for explanations of ShellCheck error codes
+
+on: [push]
```

Comment:
IMO this should be a different job under the `tests.yaml` workflow.

Also, TIL about `shellcheck`. Thanks!

Reply:
Fixed.